core/templates/pages/exploration-editor-page/preview-tab/preview-tab.component.spec.ts

```diff
@@ -0,0 +1,278 @@
+      var $rootScope = $injector.get('$rootScope');
+      $q = $injector.get('$q');
+      $uibModal = $injector.get('$uibModal');
+      contextService = $injector.get('ContextService');
+      spyOn(contextService, 'getExplorationId').and.returnValue(explorationId);
+      editableExplorationBackendApiService = $injector.get(
+        'EditableExplorationBackendApiService');
+      explorationInitStateNameService = $injector.get(
+        'ExplorationInitStateNameService');
+      explorationEngineService = $injector.get('ExplorationEngineService');
+      numberAttemptsService = $injector.get('NumberAttemptsService');
+      parameterMetadataService = $injector.get('ParameterMetadataService');
+      routerService = $injector.get('RouterService');
+      stateEditorService = $injector.get('StateEditorService');
+
+      explorationInitStateNameService.init(stateName);
+
+      spyOn(stateEditorService, 'getActiveStateName').and.returnValue(
+        stateName);
+      spyOn(parameterMetadataService, 'getUnsetParametersInfo')
+        .and.returnValue([]);
+      spyOn(editableExplorationBackendApiService, 'fetchApplyDraftExploration')
+        .and.returnValue($q.resolve(exploration));
+
+      // Mock init just to call the callback directly.
+      spyOn(explorationEngineService, 'init').and.callFake(function(
+          explorationDict, explorationVersion, preferredAudioLanguage,
+          autoTtsEnabled, successCallback) {
+        successCallback();
+      });
+
+      $scope = $rootScope.$new();
+      ctrl = $componentController('previewTab', {
+        $scope: $scope,
+        ParamChangeObjectFactory: paramChangeObjectFactory
+      });
+      ctrl.$onInit();
+    }));
+
+    it('should evaluate ctrl properties after its initialization', function() {
+      // Get data from exploration data service.
+      $scope.$apply();
+
+      expect(ctrl.isExplorationPopulated).toBe(false);
+      expect(ctrl.previewWarning).toBe('');
+    });
+
+    it('should load preview state when closing set params modal', function() {
+      spyOn($uibModal, 'open');
+      spyOn(explorationEngineService, 'initSettingsFromEditor');
+
+      // Get data from exploration data service.
+      $scope.$apply();
+      // Resolve promise in open modal.
+      $scope.$apply();
+
+      expect($uibModal.open).not.toHaveBeenCalled();
+      expect(
+        explorationEngineService.initSettingsFromEditor)
+        .toHaveBeenCalledWith(stateName, []);
+      expect(ctrl.isExplorationPopulated).toBe(true);
+    });
+
+    it('should reset preview', function() {
+      spyOn($uibModal, 'open').and.returnValue({
+        result: $q.reject()
+      });
+      spyOn(numberAttemptsService, 'reset').and.callThrough();
+      spyOn(explorationEngineService, 'initSettingsFromEditor');
+
+      // Get data from exploration data service.
+      $scope.$apply();
+      // Resolve promise in open modal.
+      $scope.$apply();
```

Comment:
Why do we need 2 $apply calls?

Reply:
We don't really need 2 `$apply` calls, I've removed it.